packages/chip/src/Chip/Chip.spec.tsx

```diff
@@ -0,0 +1,328 @@
+      });
+
+      test('renders correct label when chipCharacterLimit is 10 and chipTruncationLocation is end', () => {
+        const { chipText } = renderChip({
+          label: longLabel,
+          chipCharacterLimit: 10,
+          chipTruncationLocation: 'end',
+        });
+        expect(chipText).toHaveTextContent('crush c…');
+      });
+
+      test('renders correct label when chipCharacterLimit is 10 and chipTruncationLocation is start', () => {
+        const { chipText } = renderChip({
+          label: longLabel,
+          chipCharacterLimit: 10,
+          chipTruncationLocation: 'start',
+        });
+        expect(chipText).toHaveTextContent('…h crush');
+      });
+
+      test('renders correct label when chipCharacterLimit is 10 and chipTruncationLocation is middle', () => {
+        const { chipText } = renderChip({
+          label: longLabel,
+          chipCharacterLimit: 10,
+          chipTruncationLocation: 'middle',
+        });
+        expect(chipText).toHaveTextContent('cru…rush');
+      });
+    });
+  });
+
+  describe('dismiss button', () => {
+    test('renders when onDismiss is passed', () => {
+      const onDismiss = jest.fn();
+      const { queryByTestId } = renderChip({ onDismiss });
+      const button = queryByTestId('chip-dismiss-button');
+      expect(button).toBeInTheDocument();
+    });
+
+    test('fires onDismiss callback', () => {
+      const onDismiss = jest.fn();
+      const { queryByTestId } = renderChip({ onDismiss });
+      const button = queryByTestId('chip-dismiss-button');
+      userEvent.click(button!);
+      expect(onDismiss).toHaveBeenCalledTimes(1);
+    });
+
+    test('renders with default aria-label', () => {
+      const onDismiss = jest.fn();
+      const { queryByTestId } = renderChip({ onDismiss });
+      const button = queryByTestId('chip-dismiss-button');
+      expect(button?.getAttribute('aria-label')).toBe('Deselect chip');
+    });
+
+    test('renders with updated aria-label when dismissButtonAriaLabel is passed', () => {
+      const onDismiss = jest.fn();
+      const { queryByTestId } = renderChip({
+        onDismiss,
+        dismissButtonAriaLabel: 'new aria label',
+      });
+      const button = queryByTestId('chip-dismiss-button');
+      expect(button?.getAttribute('aria-label')).toBe('new aria label');
+    });
+
+    test('renders aria-disabled attribute when disabled is set', () => {
+      const onDismiss = jest.fn();
+      const { queryByTestId } = renderChip({ disabled: true, onDismiss });
+      const button = queryByTestId('chip-dismiss-button');
+      expect(button?.getAttribute('aria-disabled')).toBe('true');
+    });
+
+    test('does nothing when disabled button is clicked', () => {
+      const onDismiss = jest.fn();
+      const { queryByTestId } = renderChip({ disabled: true, onDismiss });
+      const button = queryByTestId('chip-dismiss-button');
+      expect(() => userEvent.click(button!)).toThrow();
```

Comment:
Curious - what does this do and how does it work?

Reply:
After upgrading to `user-event v13`, you could no longer, in tests, click on buttons with `pointer-events: none`. So instead of clicking on a button you have to test that clicking on the button will throw. However, there are other ways around [this](https://testing-library.com/docs/ecosystem-user-event/#clickelement-eventinit-options)